packages/mobile/src/hooks/useRemoteConfig.ts

```diff
@@ -1,60 +1,19 @@
-import { useMemo } from 'react'
-
-// import { createUseFeatureFlagHook } from 'audius-client/src/common/hooks/useFeatureFlag'
-// import { createUseRemoteVarHook } from 'audius-client/src/common/hooks/useRemoteVar'
-
-import {
-  AllRemoteConfigKeys,
-  BooleanKeys,
-  DoubleKeys,
-  IntKeys,
-  StringKeys,
-  FeatureFlags,
-  FeatureFlagCohortType,
-  flagCohortType
-} from 'audius-client/src/common/services/remote-config'
+import { createUseFeatureFlagHook } from 'audius-client/src/common/hooks/useFeatureFlag'
+import { createUseRemoteVarHook } from 'audius-client/src/common/hooks/useRemoteVar'
 import { getAccountUser } from 'audius-client/src/common/store/account/selectors'
 import { isRemoteConfigLoaded } from 'audius-client/src/common/store/remote-config/selectors'
 import { useSelector } from 'react-redux'
 
 import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
 import { remoteConfigInstance } from 'app/services/remote-config/remote-config-instance'
 
-// Duplicating these hooks here instead of using `createUseFeatureFlagHook` and `createUseRemoteVarHook`
-// because the react version mismatch between audius-mobile-client and audius-client is causing
-// any hooks imported from audius-client to throw an "Invalid hook call" error.
-// When the react versions are matched, this can be updated
-
-// export const useFlag = createUseFeatureFlagHook(remoteConfigInstance)
-// export const useRemoteVar = createUseRemoteVarHook(remoteConfigInstance)
-
-export const useFeatureFlag = (flag: FeatureFlags) => {
-  const configLoaded = useSelector(isRemoteConfigLoaded)
-  const userIdFlag = flagCohortType[flag] === FeatureFlagCohortType.USER_ID
-  const hasAccount = useSelectorWeb(getAccountUser)
-  const shouldRecompute = userIdFlag ? hasAccount : true
-  const isEnabled = useMemo(
-    () => remoteConfigInstance.getFeatureEnabled(flag),
-    // We want configLoaded and shouldRecompute to trigger refreshes of the memo
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [flag, configLoaded, shouldRecompute]
-  )
-  return { isLoaded: configLoaded, isEnabled }
-}
+export const useFeatureFlag = createUseFeatureFlagHook({
+  remoteConfigInstance,
+  useAccountProvider: () => !!useSelectorWeb(getAccountUser),
```

Comment:
nit: maybe `hasAccountProvider` and `hasConfigLoadedProvider`? (to indicate they're returning booleans)

Reply:
nice

Reply:
(have to include `use` otherwise hooks lint rules get mad at me)

Reply:
weiirdddd, is this technically a hook? maybe `useHasAccount` then?